### ISSUE TITLE
Duplicate key value violates unique constraint "task_instance_pkey" error

### ISSUE THREAD

**Issue by linax101** [apache/airflow#8181]
I'm using Airflow 1.10.9 with celery executor and postgres and I need to have a nested loop with the first level having 2 options and the second level going into 1800 options to loop through.

I found if I go above 600 options on the second loop I get a duplicate key value error. It seems either two sessions of postgres connects or something else is triggering the duplicate entry of my first task.

Here is an example DAG that exhibits the issue when you adjust the range from 10 to 1800 for `j`. Below should reproduce the issue.

The setup is in docker and is using this docker compose file
https://github.com/puckel/docker-airflow/blob/master/docker-compose-CeleryExecutor.yml

```
from airflow.operators.python_operator import BranchPythonOperator
from airflow.operators.dummy_operator import DummyOperator
from airflow.operators.bash_operator import BashOperator
from datetime import datetime, date, timedelta
from airflow.models import DAG
import random
import csv
from airflow.utils.dates import days_ago
from airflow.models import Variable


default_args = {
    'owner': 'airflow',
    'depends_on_past': False,
    'start_date': days_ago(2),
    'email_on_failure': False,
    'email_on_retry': False,
    'retries': 1,
    'retry_delay': timedelta(minutes=5),
}



def return_branch(**kwargs):

    branches = ['branch_0,''branch_1', 'branch_2', 'branch_3', 'branch_4']

    return random.choice(branches)

with DAG(
    dag_id='branch_demo',
    default_args=default_args,
    dagrun_timeout=timedelta(hours=2),
    schedule_interval='0 11 * * *'
) as dag:
    kick_off_dag = BashOperator(task_id='run_this_first',bash_command='echo "first"')

    branching = BranchPythonOperator(
        task_id='branching',
        python_callable=return_branch,
        provide_context=True)

    kick_off_dag >> branching

    for i in range(2):
        d = BashOperator(task_id='branch_{0}'.format(i),bash_command='echo "job"')
        for j in range(1800):
            m = BashOperator(task_id='branch_{0}_{1}'.format(i, j),bash_command='echo "done"')

            d >> m

        branching >> d
```

Postgres log error

```
ERROR:  duplicate key value violates unique constraint "task_instance_pkey"
DETAIL:  Key (task_id, dag_id, execution_date)=(run_this_first, branch_demo, 2020-04-07 18:16:14.566623+00) already exists.
STATEMENT:  INSERT INTO task_instance (task_id, dag_id, execution_date, start_date, end_date, duration, state, try_number, max_tries, hostname, unixname, job_id, pool, queue, priority_weight, operator, queued_dttm, pid, executor_config) VALUES ('run_this_first', 'branch_demo', '2020-04-07T18:16:14.566623+00:00'::timestamptz, NULL, NULL, NULL, NULL, 0, 1, '', 'root', NULL, 'default_pool', 'default', 3604, NULL, NULL, NULL, '\x80047d942e'::bytea)
```

Traceback from scheduler logs:

```
Process DagFileProcessor102313-Process:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1226, in _execute_context
    cursor, statement, parameters, context
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 854, in do_executemany
    cursor.executemany(statement, parameters)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "task_instance_pkey"
DETAIL:  Key (task_id, dag_id, execution_date)=(create_cluster, predict_model_training_combined-2, 2020-04-07 13:59:38.550927+00) already exists.
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 157, in _run_file_processor
    pickle_dags)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/db.py", line 74, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 1609, in process_file
    self._process_dags(dagbag, dags, ti_keys_to_schedule)
  File "/usr/local/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 1293, in _process_dags
    self._process_task_instances(dag, tis_out)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/db.py", line 74, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 778, in _process_task_instances
    run.verify_integrity(session=session)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/db.py", line 70, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/airflow/models/dagrun.py", line 400, in verify_integrity
    session.commit()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 1036, in commit
    self.transaction.commit()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 503, in commit
    self._prepare_impl()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 482, in _prepare_impl
    self.session.flush()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2479, in flush
    self._flush(objects)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2617, in _flush
    transaction.rollback(_capture_exception=True)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 153, in reraise
    raise value
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2577, in _flush
    flush_context.execute()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/unitofwork.py", line 422, in execute
    rec.execute(self)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/unitofwork.py", line 589, in execute
    uow,
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/persistence.py", line 245, in save_obj
    insert,
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/persistence.py", line 1084, in _emit_insert_statements
    c = cached_connections[connection].execute(statement, multiparams)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 982, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 293, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1101, in _execute_clauseelement
    distilled_params,
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1250, in _execute_context
    e, statement, parameters, cursor, context
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1476, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 152, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1226, in _execute_context
    cursor, statement, parameters, context
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 854, in do_executemany
    cursor.executemany(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "task_instance_pkey"
DETAIL:  Key (task_id, dag_id, execution_date)=(create_cluster, predict_model_training_combined-2, 2020-04-07 13:59:38.550927+00) already exists.
[SQL: INSERT INTO task_instance (task_id, dag_id, execution_date, start_date, end_date, duration, state, try_number, max_tries, hostname, unixname, job_id, pool, queue, priority_weight, operator, queued_dttm, pid, executor_config) VALUES (%(task_id)s, %(dag_id)s, %(execution_date)s, %(start_date)s, %(end_date)s, %(duration)s, %(state)s, %(try_number)s, %(max_tries)s, %(hostname)s, %(unixname)s, %(job_id)s, %(pool)s, %(queue)s, %(priority_weight)s, %(operator)s, %(queued_dttm)s, %(pid)s, %(executor_config)s)]
[parameters: ({'task_id': 'create_cluster', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 7119, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd653458570>}, {'task_id': 'before_training', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 7118, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd6534582d0>}, {'task_id': 'after_training', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 1, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd653458540>}, {'task_id': 'remove_cluster', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 1, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd653458330>}, {'task_id': 'train_model_AGGH_eon', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 3, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd653458180>}, {'task_id': 'watch_training_AGGH_eon', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 2, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd653458360>}, {'task_id': 'train_model_AGGH_ein', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 3, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd653458630>}, {'task_id': 'watch_training_AGGH_ein', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 2, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd6534586f0>}  ... displaying 10 of 7120 total bound parameter sets ...  {'task_id': 'train_model_ZYYJ_ein', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 3, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd652758660>}, {'task_id': 'watch_training_ZYYJ_ein', 'dag_id': 'predict_model_training_combined-2', 'execution_date': datetime.datetime(2020, 4, 7, 13, 59, 38, 550927, tzinfo=<TimezoneInfo [UTC, GMT, +00:00:00, STD]>), 'start_date': None, 'end_date': None, 'duration': None, 'state': None, 'try_number': 0, 'max_tries': 1, 'hostname': '', 'unixname': 'root', 'job_id': None, 'pool': 'default_pool', 'queue': 'default', 'priority_weight': 2, 'operator': None, 'queued_dttm': None, 'pid': None, 'executor_config': <psycopg2.extensions.Binary object at 0x7fd6527586f0>})]
(Background on this error at: http://sqlalche.me/e/gkpj)
Process DagFileProcessor104982-Process:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1226, in _execute_context
    cursor, statement, parameters, context
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 854, in do_executemany
    cursor.executemany(statement, parameters)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "task_instance_pkey"
DETAIL:  Key (task_id, dag_id, execution_date)=(create_cluster, predict_model_training_combined-4, 2020-04-07 14:41:02.111727+00) already exists.
```

**Comment by linax101** [comment]
NOTE: SequentialExecuter is not affected. Local and Celery are. I'm trying Celery again with less parallelism in the config.

**Comment by mik-laj** [comment]
Have you checked whether changing option ``max-tis-per-query`` affects this problem?
https://airflow.readthedocs.io/en/latest/configurations-ref.html#max-tis-per-query

**Comment by linax101** [comment]
I tried changing it from 512 to 10000 to 0 (for unlimited) and i still have the same problem.

**Comment by mik-laj** [comment]
I have no idea what might be causing this problem. Do you want to work on this ticket? Airflow is an Open Source project, so there is no paid technical support. Each problem is solved by other users who have a similar problem.

Thank you very much for reporting the error and the exact description.

**Comment by linax101** [comment]
Yes i would like to, we are trying to implement airflow for model training. It would work with SequentialExecuter but that would take forever to get through this many tasks. I'm experimenting with adding a sleep task or two and adjusting concurrency per task. So far it makes the postgres error go away but tasks are stuck in a None state and not queuing or starting/scheduling.

**Comment by mik-laj** [comment]
If the tasks are in the None state then the problem exists somewhere in this block of code.
https://github.com/apache/airflow/blob/master/airflow/jobs/scheduler_job.py#L861-L863
1 line create task instance with none state
3 line change state from none to SCHEDULED

**Comment by mik-laj** [comment]
If you want to see increase verbosity in logs then run the following command: 
```bash
export AIRFLOW__CORE__LOGGING_LEVEL=debug
```

**Comment by linax101** [comment]
Yeah i have debug enabled. Not seeing anything stand out.

**Comment by linax101** [comment]
I'll capture some debug logs and upload them here to see if you spot anything out of the ordinary.

**Comment by linax101** [comment]
I discovered that 1.10.2 is not affected by my problem. I went through several versions going back from 1.10.9 and each one had the same issue until i got to 1.10.2. If I lower the parallelism 1.10.2 does not get the database error, it still gets the error at higher configs of parallelism but somehow the tasks still progresses.

Attached are the logs surrounding my attempt at running my test DAG along with the pkey failure error.

[airflow.log](https://github.com/apache/airflow/files/4452601/airflow.log)

**Comment by ghostbody** [comment]
We met another similar problem when using `RestApi` `POST /api/experimental/dags/<DAG_ID>/dag_runs` to trigger a DAG Run.

However there is a unique key for `(dag_id, execution_date)` and `execution_date` is `timestamp` (i.e. ms) level. When trigger multi DAG Runs programatically, it's likely to get two DAG Runs with the same `execution_date` and `duplicate key error` will occur.

I think `execution_date` is a decent but not a good design to identify the DAG Run's uniqueness. 🤔 

In the other hand, the mysql `id` is more reasonable in some certain scenes.

**Comment by linax101** [comment]
As a update, latest 1.10.10 is still affected by this problem.

**Comment by edi33416** [comment]
Any updates on this issue?
Version 1.10.12 is still affected by this

**Comment by ashb** [comment]
Can someone test this on 2.0.0 please?

**Comment by kaxil** [comment]
Going to close this one. If someone reproduces this with 2.0, we can reopen it.

Thanks

**Comment by DreamyWen** [comment]
i encounter this in 2.1.1 airflow

2021-08-15 00:31:16 [pool-1-thread-1] INFO  c.s.e.bi.executor.MainExecutor - 等待dag script_dag jobId ad9435cd-459e-40c3-b9ef-2ee654641b06 执行完毕..
2021-08-15 00:31:16 [pool-1-thread-1] INFO  c.s.e.bi.executor.MainExecutor - python_info: /data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/configuration.py:346 DeprecationWarning: The hide_sensitive_variable_fields option in [admin] has been moved to the hide_sensitive_var_conn_fields option in [core] - the old setting has been used, but please update your config./data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/configuration.py:346 DeprecationWarning: The default_queue option in [celery] has been moved to the default_queue option in [operators] - the old setting has been used, but please update your config./data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/configuration.py:346 DeprecationWarning: The default_queue option in [celery] has been moved to the default_queue option in [operators] - the old setting has been used, but please update your config.[[34m2021-08-15 00:31:16,050[0m] {[34m__init__.py:[0m38} INFO[0m - Loaded API auth backend: <module 'airflow.api.auth.backend.deny_all' from '/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/api/auth/backend/deny_all.py'>[0m/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/plugins_manager.py:239 DeprecationWarning: This decorator is deprecated.In previous versions, all subclasses of BaseOperator must use apply_default decorator for the`default_args` feature to work properly.In current version, it is optional. The decorator is applied automatically using the metaclass.Traceback (most recent call last):  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context    self.dialect.do_execute(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute    cursor.execute(statement, parameters)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/cursors.py", line 255, in execute    self.errorhandler(self, exc, value)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/connections.py", line 50, in defaulterrorhandler    raise errorvalue  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/cursors.py", line 252, in execute    res = self._query(query)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/cursors.py", line 378, in _query    db.query(q)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/connections.py", line 280, in query    _mysql.connection.query(self, query)_mysql_exceptions.IntegrityError: (1062, "Duplicate entry 'script_dag-2021-08-14 16:31:16.000000' for key 'dag_id'")The above exception was the direct cause of the following exception:Traceback (most recent call last):  File "/data/anaconda3/envs/airflow/bin/airflow", line 8, in <module>    sys.exit(main())  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/__main__.py", line 40, in main    args.func(args)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/cli/cli_parser.py", line 48, in command    return func(*args, **kwargs)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/utils/cli.py", line 91, in wrapper    return f(*args, **kwargs)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/cli/commands/dag_command.py", line 129, in dag_trigger    message = api_client.trigger_dag(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/api/client/local_client.py", line 29, in trigger_dag    dag_run = trigger_dag.trigger_dag(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/api/common/experimental/trigger_dag.py", line 117, in trigger_dag    triggers = _trigger_dag(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/api/common/experimental/trigger_dag.py", line 83, in _trigger_dag    trigger = _dag.create_dagrun(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/utils/session.py", line 70, in wrapper    return func(*args, session=session, **kwargs)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/airflow/models/dag.py", line 1796, in create_dagrun    session.flush()  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2540, in flush    self._flush(objects)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2682, in _flush    transaction.rollback(_capture_exception=True)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py", line 68, in __exit__    compat.raise_(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_    raise exception  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2642, in _flush    flush_context.execute()  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 422, in execute    rec.execute(self)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/orm/unitofwork.py", line 586, in execute    persistence.save_obj(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 239, in save_obj    _emit_insert_statements(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/orm/persistence.py", line 1135, in _emit_insert_statements    result = cached_connections[connection].execute(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1011, in execute    return meth(self, multiparams, params)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection    return connection._execute_clauseelement(self, multiparams, params)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1124, in _execute_clauseelement    ret = self._execute_context(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context    self._handle_dbapi_exception(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception    util.raise_(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_    raise exception  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context    self.dialect.do_execute(  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute    cursor.execute(statement, parameters)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/cursors.py", line 255, in execute    self.errorhandler(self, exc, value)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/connections.py", line 50, in defaulterrorhandler    raise errorvalue  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/cursors.py", line 252, in execute    res = self._query(query)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/cursors.py", line 378, in _query    db.query(q)  File "/data/anaconda3/envs/airflow/lib/python3.8/site-packages/MySQLdb/connections.py", line 280, in query    _mysql.connection.query(self, query)sqlalchemy.exc.IntegrityError: (_mysql_exceptions.IntegrityError) (1062, "Duplicate entry 'script_dag-2021-08-14 16:31:16.000000' for key 'dag_id'")[SQL: INSERT INTO dag_run (dag_id, execution_date, start_date, end_date, state, run_id, creating_job_id, external_trigger, run_type, conf, last_scheduling_decision, dag_hash) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)][parameters: ('script_dag', datetime.datetime(2021, 8, 14, 16, 31, 16), datetime.datetime(2021, 8, 14, 16, 31, 16, 165205), None, 'running', 'ad9435cd-459e-40c3-b9ef-2ee654641b06', None, 1, <DagRunType.MANUAL: 'manual'>, b'\x80\x05\x95\xad\x03\x00\x00\x00\x00\x00\x00}\x94(\x8c\x04key1\x94X\xc0\x02\x00\x00{"biz_type":1,"create_time":"2021-08-1423:14:03","create_user_cn" ... (1057 characters truncated) ... 94\x8c\x04key3\x94\x8c\x06online\x94\x8c\tuse_queue\x94\x8c\x1aonline_script_10.41.143.17\x94\x8c\x0bredis_queue\x94\x8c\x13airflow_eval_status\x94u.', None, '21462d7982e52b0a4c60655939353519')](Background on this error at: http://sqlalche.me/e/13/gkpj)
2021-08-15 00:31:16 [pool-1-thread-1] INFO  c.s.e.b.s.service.AirflowService - 此次请求，本机ip[10.41.1

**Comment by tnyz** [comment]
we noticed this issue as well in airflow 2.3.3
```
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "task_instance_pkey"
80
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(ingestion-scheduler, check_for_success_flag, manual__2022-10-03T15:56:03.241451+00:00, 0) already exists.
```

**Comment by JasperSui** [comment]
> we noticed this issue as well in airflow 2.3.3
> 
> ```
> psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "task_instance_pkey"
> 80
> DETAIL:  Key (dag_id, task_id, run_id, map_index)=(ingestion-scheduler, check_for_success_flag, manual__2022-10-03T15:56:03.241451+00:00, 0) already exists.
> ```

Same here.

**Comment by LucaSoato** [comment]
Same issue here, is there a solution?

**Comment by potiuk** [comment]
> Same issue here, is there a solution?

As soon as someone provides more information coming from the latest version of Airlfow ideally - including logs and circumstances - versions etc.  (or ideally opens a new issue with all those details) I doubt there wil be solution @JasperSui @LucaSoato @tnyz. 

I suggest to update to latest released version of Airflow and if the issue is there - report it (again ideally as a new issue with stacktraces and all details and linked to that isssue as likely similar). Adding "I also have the same problem" is good information but without extra details and stacktraces from the latest version it does not bring us any closer to finding a solution. 

Please try to help the community by providing those details - this is the only 

We implement hundreds of fixes in each minor release. And it's possible the problem has been fixed there. We ONLY release fixes in the latest active branch - so you will have to upgrade to latest 2.* version anyway to get a fix (if there will be one) - upgrading now and reporting if the error is still there is the most efficient way how to speed up solving the issue. 

@JasperSui @LucaSoato @tnyz - can we count on your help to the community with this one?

**Comment by JasperSui** [comment]
> > Same issue here, is there a solution?
> 
> 
> 
> As soon as someone provides more information coming from the latest version of Airlfow ideally - including logs and circumstances - versions etc.  (or ideally opens a new issue with all those details) I doubt there wil be solution @JasperSui @LucaSoato @tnyz we can try to diagnose and fix it.
> 
> 
> 
> I suggest to update to latest released version of Airflow and if the issue is there - report it (again ideally as a new issue with stacktraces and all details and linked to that isssue as likely similar). Adding "I also have the same problem" is good information but without extra details and stacktraces from the latest version it does not bring us any closer to finding a solution. 
> 
> 
> 
> Please try to help the community by providing those details - this is the only 
> 
> 
> 
> We implement hundreds of fixes in each minor release. And it's possible the problem has been fixed there. We ONLY release fixes in the latest active branch - so you will have to upgrade to latest 2.* version anyway to get a fix (if there will be one) - upgrading now and reporting if the error is still there is the most efficient way how to speed up solving the issue. 
> 
> 
> 
> @JasperSui @LucaSoato @tnyz - can we count on your help to the community with this one? 

Yeah, I upgraded to 2.4.1 and it doesn't happen again, I believe upgrading to 2.4.* will fix this.

**Comment by Sangameshmsopen** [comment]
Hello all,
We have observed that in Airflow scheduler heartbeat has stopped and all the DAGs were in queued or running state. We have checked scheduler logs and got the below error. May i know what might be the cause for this.
Details: We are running Airflow in Google cloud Composer service with below versions
composer-2.1.4
airflow-2.4.3
 Thanks in advance
 
 sqlalchemy.exc.PendingRollbackError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "task_instance_pkey"
